### DISCUSSION TITLE
Prevent Collapse button from launch the modal

### DISCUSSION THREAD

**Discussion by Feriksson** [rappasoft/laravel-livewire-tables#1660] [Q&A] [answered]
Hello, i need to prevent the 'collapse button' to not tigger my modal.

I followed the instruction from the modal documentation itselft, on the sponsored area.

It's seems like when i configured into the configure method:

 `$this->setPrimaryKey('id')
        ->setTableRowUrl(fn () => '#view')
        ->setTrAttributes(function ($row) {
            return [
                'wire:click' => 'viewHistoryModal('.$row->id.')'
            ];
        });`

But, now it's seem like the componente will ignore the column even with **->unclickable()** instruction explicit on it, and the collapse button will also pop the modal too.

How i should disable this behavior, i apreciate any help or orientation.

https://github.com/rappasoft/laravel-livewire-tables/assets/10608134/6480b097-757c-496e-9d06-f9d263117022

**Comment by lrljoe** [comment, marked as answer]
At the moment, you're using setTrAttributes, which is run **before** the columns are generated, as it sets the attributes on the <tr> before the <td> elements are created.

If you use setTdAttributes, then you can test if the Column is unclickable, before appending the attributes.

**Comment by Feriksson** [comment]
Hello lrljoe, thanks for the aproach, this has solved the problem so far, thanks.
